src/entity/user/referral/UserReferral.ts

```diff
@@ -39,6 +44,11 @@ export class UserReferral {
   })
   userId: string;
 
+  @Column({
+    type: 'text',
+  })
+  externalUserId: string;
+
```

Comment:
@omBratteng introduced a property instead as it must be present at all times. Much easier to manage. Then the URL is kept in the flags.